landoui/revisions.py

```diff
@@ -277,6 +277,31 @@ def sec_approval_request_handler():
     return jsonify({})
 
 
+@revisions.route("/landing_jobs/<int:landing_job_id>/cancel", methods=("PUT",))
+def cancel_landing_job(landing_job_id):
+    if not is_user_authenticated():
+        errors = make_form_error("You must be logged in to update a landing job.")
+        return jsonify(errors=errors), 401
+
+    token = get_phabricator_api_token()
+    api = LandoAPI(
+        current_app.config["LANDO_API_URL"],
+        auth0_access_token=session.get("access_token"),
+        phabricator_api_token=token,
+    )
+
+    try:
+        data = api.request(
```

Comment:
Is it common to have an API endpoint in lando-ui that wraps a call to an API endpoint in lando-api? Seems like we could call lando-api directly or something, no?

Also a little funky that we pass an `action` to lando-api via the body but we use `/cancel` in the route for lando-ui. Again, I'm not sure if this is how things are usually done in the Lando internals.

Reply:
> Is it common to have an API endpoint in lando-ui that wraps a call to an API endpoint in lando-api? Seems like we could call lando-api directly or something, no?

Yes it is common -- it was part of the original design decision to separate the two in this way, and these endpoints in lando-api are not accessible from the public API. As we move towards a more API driven UI, it will make sense to consolidate these endpoints into lando-api, but there are some security and auth implications that we will have to address.

>Also a little funky that we pass an action to lando-api via the body but we use /cancel in the route for lando-ui. Again, I'm not sure if this is how things are usually done in the Lando internals.

Agreed -- There isn't an established convention yet for the concept of "actions", though this could probably be changed to a generic RESTful PUT endpoint that takes the status as a parameter. I will rewrite these to make them more consistent/generic.